neo/Core/TX/Transaction.py

```diff
@@ -683,7 +688,7 @@ def Verify(self, mempool):
     #                    return False
     #
 
-    def GetScriptHashesForVerifying(self):
+    def GetScriptHashesForVerifying(self, snapshot):
```

Comment:
@ixje adding `snapshot` as a required input here breaks the current `RawTransaction` class because it requires a db instance. I don't think this parameter should be required, since it not required in the case of a `raw_tx` as seen below.

Reply:
I'd say `snapshot` is required for correct default NEO behaviour. However, I think we can make an exception here to accommodate for your raw_tx class. Please PR to make it optional.

Reply:
@ixje Please see https://github.com/CityOfZion/neo-python/pull/981